sdk/python/kfp/components/executor.py

```diff
@@ -98,8 +101,64 @@ def _get_input_artifact_path(self, artifact_name: str):
               artifact_name))
     return input_artifact.path
 
-  def _write_executor_output(self):
-    executor_output = {'artifacts': {}}
+  def _write_output_parameter_value(self, name: str, value: Union[str, int,
+                                                                  float]):
+    if type(value) == str:
+      output = {"stringValue": value}
+    elif type(value) == int:
+      output = {"intValue": value}
+    elif type(value) == float:
+      output = {"doubleValue": value}
+    else:
+      raise RuntimeError(
+          'Expected value of type str, int or float, got {} instead for value {}'
+          .format(type(value), value))
+
+    if not self._executor_output.get('parameters'):
+      self._executor_output['parameters'] = {}
+
+    self._executor_output['parameters'][name] = output
+
+  def _write_output_artifact_payload(self, name: str, value: Any):
+    path = self._get_output_artifact_path(name)
+    with open(path, 'w') as f:
+      f.write(str(value))
+
+  @classmethod
+  def _is_parameter(cls, annotation: Any) -> bool:
+    return annotation in [str, int, float]
+
+  @classmethod
+  def _is_artifact(cls, annotation: Any) -> bool:
+    if type(annotation) == type:
+      return issubclass(annotation, Artifact)
```

Comment:
How does this work for custom artifact types?

Reply:
It's not supported right now.